wrapper/src/test/java/integration/container/aurora/postgres/AuroraPostgresReadWriteSplittingPerformanceTest.java

```diff
@@ -0,0 +1,354 @@
+        }
+
+        row = sheet.createRow(rows + 1);
+        perfStat.writeData(row);
+      }
+
+      // Write to file
+      final File newExcelFile = new File(fileName);
+      newExcelFile.createNewFile();
+      try (FileOutputStream fileOut = new FileOutputStream(newExcelFile)) {
+        workbook.write(fileOut);
+      }
+    }
+  }
+
+  protected static Properties initNoPluginPropsWithTimeouts() {
+    final Properties props = new Properties();
+    props.setProperty(PGProperty.USER.getName(), AURORA_POSTGRES_USERNAME);
+    props.setProperty(PGProperty.PASSWORD.getName(), AURORA_POSTGRES_PASSWORD);
+    props.setProperty(PGProperty.TCP_KEEP_ALIVE.getName(), Boolean.FALSE.toString());
+    props.setProperty(PGProperty.CONNECT_TIMEOUT.getName(), "30");
+    props.setProperty(PGProperty.SOCKET_TIMEOUT.getName(), "30");
+
+    return props;
+  }
+
+  protected static Properties initReadWritePluginProps() {
+    final Properties props = initNoPluginPropsWithTimeouts();
+    props.setProperty(PropertyDefinition.PLUGINS.name, "auroraHostList,readWriteSplitting");
+    props.setProperty(ReadWriteSplittingPlugin.LOAD_BALANCE_READ_ONLY_TRAFFIC.name, "true");
+    return props;
+  }
+
+  Result getSetReadOnlyResults(final Properties props)
+      throws SQLException {
+    Long setReadOnlyTrueStartTime;
+    final List<Long> elapsedSetReadOnlyTrueTimes = new ArrayList<>(REPEAT_TIMES);
+    Long setReadOnlyFalseStartTime;
+    final List<Long> elapsedSetReadOnlyFalseTimes = new ArrayList<>(REPEAT_TIMES);
+    final Result result = new Result();
+
+    for (int i = 0; i < REPEAT_TIMES; i++) {
+
+      try (final Connection conn = connectToInstance(POSTGRES_CLUSTER_URL, AURORA_POSTGRES_PORT, props)) {
+
+        setReadOnlyTrueStartTime = System.nanoTime();
+        conn.setReadOnly(true);
+        final long setReadOnlyTrueElapsedTime = (System.nanoTime() - setReadOnlyTrueStartTime);
+        elapsedSetReadOnlyTrueTimes.add(setReadOnlyTrueElapsedTime);
+
+        setReadOnlyFalseStartTime = System.nanoTime();
+        conn.setReadOnly(false);
+        final long setReadOnlyFalseElapsedTime = (System.nanoTime() - setReadOnlyFalseStartTime);
+        elapsedSetReadOnlyFalseTimes.add(setReadOnlyFalseElapsedTime);
+      }
+    }
+    final long setReadOnlyTrueMin = elapsedSetReadOnlyTrueTimes.stream().min(Long::compare).orElse(0L);
+    final long setReadOnlyTrueMax = elapsedSetReadOnlyTrueTimes.stream().max(Long::compare).orElse(0L);
+    final long setReadOnlyTrueAvg =
+        (long) elapsedSetReadOnlyTrueTimes.stream().mapToLong(a -> a).summaryStatistics().getAverage();
+
+    final long setReadOnlyFalseMin = elapsedSetReadOnlyFalseTimes.stream().min(Long::compare).orElse(0L);
+    final long setReadOnlyFalseMax = elapsedSetReadOnlyFalseTimes.stream().max(Long::compare).orElse(0L);
+    final long setReadOnlyFalseAvg =
+        (long) elapsedSetReadOnlyFalseTimes.stream().mapToLong(a -> a).summaryStatistics().getAverage();
```

Comment:
Will you get the same result by storing the summaryStatistics first then just calling getMax(), getMin() and getAverage()?
```java
    final LongSummaryStatistics stats = elapsedSetReadOnlyFalseTimes.stream().mapToLong(a -> a).summaryStatistics();
    stats.getMax();
    stats.getMin();
    stats.getAverage();
```